aten/CLAUDE.md

```diff
@@ -0,0 +1,66 @@
+# ATen/ Directory - C++ Tensor Library
+
+ATen (A Tensor library) is PyTorch's C++ tensor computation library providing the core backend for all tensor operations.
+
+## 🏗️ Directory Organization
+
+### Core Components
+- **`src/ATen/`** - Main ATen library with tensor operations and kernels
+- **`src/ATen/core/`** - Core tensor types, dispatch system, and utilities
+- **`src/ATen/native/`** - Native kernel implementations for CPU/CUDA/other backends
+- **`src/ATen/templates/`** - Code generation templates used by torchgen
+- **`src/README.md`** - Reference counting and low-level tensor library documentation
+
+### Backend Implementations
+- **`src/ATen/cpu/`** - CPU-specific implementations and vectorization
+- **`src/ATen/cuda/`** - CUDA GPU implementations and utilities
+- **`src/ATen/mps/`** - Apple Metal Performance Shaders backend
+- **`src/ATen/hip/`** - AMD ROCm/HIP backend support
+- **`src/ATen/vulkan/`** - Vulkan compute backend
+- **`src/ATen/xpu/`** - Intel XPU backend support
+
+### Key Subsystems
+- **`src/ATen/functorch/`** - Functional transforms (vmap, grad) C++ implementation
+- **`src/ATen/quantized/`** - Quantized tensor operations
+- **`src/ATen/ops/`** - Operator interface definitions
+- **`src/ATen/test/`** - Comprehensive C++ test suite
+
+## 🔧 Key Files & Concepts
+
+### Core Headers
+- `src/ATen/ATen.h` - Main ATen header with all tensor operations
+- `src/ATen/Tensor.h` - Tensor class definition
+- `src/ATen/TensorIterator.h` - Efficient tensor iteration framework
+- `src/ATen/core/Tensor.h` - Core tensor interface
+- `src/ATen/Dispatch.h` - Type dispatch system for kernels
+
+### Per-Operator Headers
+Use `#ifndef AT_PER_OPERATOR_HEADERS` pattern to conditionally include individual operator headers vs. monolithic headers for faster compilation.
+
+### Code Generation
+- `src/ATen/native/native_functions.yaml` - Operator definitions for codegen
+- `src/ATen/templates/` - Jinja2 templates for generating C++ code
+- Generated files appear in `build/aten/src/ATen/` directory after build
+
+### Testing
+```bash
+# Test specific kernel implementations
+python test/run_test.py -i test_ops
```

Comment:
stuff I don't want my agent to do

Reply:
this file takes so long to run! agents will waste all its time waiting for the tests to finish